gensim/models/translation_matrix.py

```diff
@@ -393,6 +393,9 @@ def __init__(self, tagged_docs, source_lang_vec, target_lang_vec, random_state=N
         self.random_state = utils.get_random_state(random_state)
         self.translation_matrix = None
 
+        if tagged_docs is not None:
```

Comment:
Problem isn't fixed in `train` method (you continue to ignore passed parameter).